courses/api.py

```diff
@@ -50,6 +56,48 @@
 )
 
 
+def get_user_relevant_course_run(
+    course: Course, user: Optional[User], now: Optional[datetime] = None
+) -> CourseRun:
+    """
+    For a given Course, finds the course run that is the most relevant to the user.
+    For anonymous users, this means the soonest enrollable course run.
+    For logged-in users, this means an active course run that they're enrolled in, or the soonest enrollable course run.
+    """
+    now = now or now_in_utc()
+    run_qset = (
+        course.courseruns.exclude(start_date=None)
+        .exclude(enrollment_start=None)
+        .filter(Q(end_date=None) | Q(end_date__gt=now))
+    )
+    if user and user.is_authenticated:
+        user_enrollments = Count(
+            "enrollments",
+            filter=Q(
+                enrollments__user=user,
+                enrollments__active=True,
+                enrollments__edx_enrolled=True,
+            ),
+        )
+        run_qset = run_qset.annotate(user_enrollments=user_enrollments).order_by(
+            "-user_enrollments", "enrollment_start"
+        )
+        runs = (
+            run
+            for run in run_qset
+            if run.user_enrollments > 0
+            or (run.enrollment_end is None or run.enrollment_end > now)
```

Comment:
Probably we want to use `run.enrollments` based on the current user because the scenario is currently failing based on the below scenario:
1. Enroll a user in a Course Run(e.g Run1)
2. Set the start date of the Run1 in past
3. Set the end date of the Run1 in future
4. set enrollment start & end date in past

In the above scenario, the `Enrolled` button or any button doesn't show. However, in this case, the current run of the user is still in progress and `Enrolled` button should show and link the user to the course.

Reply:
@arslanashraf7 That's what this PR was designed to do. I think your enrollment record might not have `active=True` or `edx_enrolled=True`. In either case, the query on the backend would ignore your enrollment in the in-progress course run since we shouldn't be linking you to edX if your enrollment is inactive or it didn't go through in edX. I just tested locally and it's working as expected. If that doesn't work for you, or if I'm not understanding you correctly, let me know.

Reply:
You are correct, Thanks for responding with details. In my case  I somehow unchecked the `edx_enroled` for the course run and due to this, I was facing the above issue. But correcting that makes the uses cases work as well.